kiota_http/middleware/options/redirect_handler_option.py

```diff
@@ -13,7 +13,7 @@ class RedirectHandlerOption(RequestOption):
 
     def __init__(
         self,
-        max_redirect: int = DEFAULT_MAX_REDIRECT,
+        max_redirect: int = None,
```

Comment:
Out of curiosity, should this be a `0` or can `None` be passed to an int?
```suggestion
        max_redirect: int = 0,
```